Root/OptimizationDump.cxx

```diff
@@ -377,13 +377,17 @@ EL::StatusCode OptimizationDump :: execute ()
   }
 
   if(!m_inputMET.empty() && !m_inputJets.empty()){
-    m_effectiveMass = VD::Meff_inclusive(in_met, in_jets, in_muons, in_electrons)/1000.;
+    
+    ConstDataVector<xAOD::JetContainer> presel_jets = VD::subset_using_decor(in_jets, VD::decor_vd_pass_Presel, 1);
+    //std::cout << in_jets->size() << " " << presel_jets.size() << std::endl;
+    m_effectiveMass = VD::Meff_inclusive(in_met, presel_jets.asDataVector(), in_muons, in_electrons)/1000.;
```

Comment:
Ok, yes, this is what I was getting at. Leptons do this too— by vetoing on cosmics and bads— as you point out.

On Oct 15, 2015, at 9:24 PM, Giordon Stark <notifications@github.com<mailto:notifications@github.com>> wrote:


In Root/OptimizationDump.cxx<https://github.com/kratsg/TheAccountant/pull/136#discussion_r42203447>:

> @@ -377,13 +377,17 @@ EL::StatusCode OptimizationDump :: execute ()
>    }
>
>    if(!m_inputMET.empty() && !m_inputJets.empty()){
> -    m_effectiveMass = VD::Meff_inclusive(in_met, in_jets, in_muons, in_electrons)/1000.;
> +
> +    ConstDataVector<xAOD::JetContainer> presel_jets = VD::subset_using_decor(in_jets, VD::decor_vd_pass_Presel, 1);
> +    //std::cout << in_jets->size() << " " << presel_jets.size() << std::endl;
> +    m_effectiveMass = VD::Meff_inclusive(in_met, presel_jets.asDataVector(), in_muons, in_electrons)/1000.;


Actually, it makes a lot more sense to have it cut on the effective mass calculated from jets that pass the preselection -- just like we would calculate mT from leptons that pass the preselection. Both of these can be cut on in a clear and consistent way that makes sense from the perspective of a preselection or using SR or CR definitions.

—
Reply to this email directly or view it on GitHub<https://github.com/kratsg/TheAccountant/pull/136/files#r42203447>.